submissions/dafen173/a-tiny-js-world2/index.js

```diff
@@ -0,0 +1,92 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+class Person {
+  constructor(gender, name, species) {
+    this.species = species;
+    this.gender = gender;
+    this.name = name;  
+  } 
+  setFriends(friends){
+    this.friends = friends;
+  } 
+  
+  sayAsAnimal(){
+    if (this.species === 'cat') {
+      this.speach = 'Meow';
+    } else if (this.species === 'cat-woman') {
+      this.speach = 'Meow'
+    } else {
+      this.speach = 'Wohw whow whow'
+    }
```

Comment:
Create classes for every species and let them handle what they do.
A base class should not know anything about child classes implementation.
Imagine, you have dozens of species. Current approach will lead to this method growing enermously becoming heavy with `if/else` constructions. No need - let every species class handle this.

Method overloading would come to rescue, but once you have a class per species you won't need this.